packages/tabs/src/Tabs.tsx

```diff
@@ -0,0 +1,297 @@
+  if (!(element != null && typeof element === 'object' && 'type' in element)) {
+    return false;
+  }
+
+  return (element.type as any).displayName === 'Tab';
+}
+
+interface TabsProps {
+  /**
+   * Content that will appear inside of Tabs component. Should be comprised of at least two Tabs.
+   */
+  children: Array<React.ReactElement>;
+
+  /**
+   * Callback to be executed when Tab is selected. Receives index of activated Tab as the first argument.
+   */
+  setSelected?: any;
+
+  /**
+   * Index of the Tab that should appear active. If value passed to selected prop, component will be controlled by consumer.
+   */
+  selected?: number;
+
+  /**
+   * className supplied to Tabs container.
+   */
+  className?: string;
+
+  /**
+   * HTML Element that wraps title in Tab List.
+   */
+  as?: React.ElementType<any>;
+}
+
+/**
+ * # Tabs
+ *
+ * Tabs component
+ *
+ * ```
+<Tabs selected={0} setSelected={() => execute callback when new Tab is selected}>
+  <Tab name='First Tab'>Tab 1</Tab>
+  <Tab name='Second Tab'>Tab 2</Tab>
+</Tabs>
+```
+ * @param props.children Content to appear inside of Tabs component.
+ * @param props.setSelected Callback to be executed when Tab is selected. Receives index of activated Tab as the first argument.
+ * @param props.selected Index of the Tab that should appear active. If value passed, component will be controlled by consumer.
+ * @param props.className className applied to Tabs container.
+ * @param props.as HTML Element that wraps name in Tab List.
+ */
+function Tabs({
+  children,
+  setSelected,
+  selected,
+  className,
+  as = 'button',
+  ...rest
+}: TabsProps) {
+  const isControlled = typeof selected === 'number';
+  const childrenArray = React.Children.toArray(children);
+  const [activeIndex, setActiveIndex] = useState(
+    childrenArray.findIndex(child => child.props.default) || 0,
+  );
+  const tabListRef = useRef<HTMLDivElement>(null);
+  const [focusedState, setFocusedState] = useState([0]);
+
+  const currentIndex = childrenArray.findIndex((child, index) => {
+    if (!child) {
+      return false;
+    }
+
+    if (typeof activeIndex === 'number' || isControlled) {
+      return [activeIndex, selected].includes(index);
+    }
+
+    return child.props.default;
+  });
+
+  function handleChange(
+    e: React.SyntheticEvent<Element, MouseEvent>,
+    index: number,
+  ) {
+    if (!isControlled && !selected) {
+      setActiveIndex(index);
+    }
+
+    if (setSelected) {
+      setSelected(index);
+    }
+  }
+
+  function handleKeyDown(e: React.KeyboardEvent) {
+    e.stopPropagation();
+
+    const enabledIndexes = childrenArray.reduce(
+      (acc, child, index) => {
+        if (child.props.disabled) {
+          return acc;
+        }
+
+        return [...acc, index];
+      },
+      [] as Array<number>,
+    );
+
+    const enabledCurrentIndex = enabledIndexes.indexOf(
+      isControlled ? (selected as number) : activeIndex,
```

Comment:
Alternatively you can do `selected!` to tell TS that you know this value isn't undefined here. 

[Docs here](https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#non-null-assertion-operator)

Reply:
👍